### PR TITLE
karmada-agent: move all default values to NewOptions

### PR DESCRIPTION
**Pull request by Zhuzhenghao**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
I move all default options to function NewOptions ，so we can get all default values easily

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by XiShanYongYe-Chang** [comment]
Hi @Zhuzhenghao Welcome to join the karmada. You can sign off refer to https://github.com/karmada-io/karmada/pull/2878/checks?check_run_id=9742553922

**Comment by Zhuzhenghao** [comment]
@XiShanYongYe-Chang Done. Can you approve running workflows?

**Comment by Zhuzhenghao** [comment]
@XiShanYongYe-Chang I think this change is reasonable.

FYI: https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-apiserver/app/options/options.go

**Review by XiShanYongYe-Chang — COMMENTED** [review]
/cc @lonelyCZ @RainbowMango

**Comment by lonelyCZ** [comment]
Look good. We can find default options quickly by setting them in `NewOption` and make the code clearer. I also see this usage in other packages.

File: cmd/agent/app/options/options.go [hunk 1]
```diff
@@ -127,11 +115,37 @@ type Options struct {
...
 		LeaderElection: componentbaseconfig.LeaderElectionConfiguration{
 			LeaderElect:       true,
 			ResourceLock:      resourcelock.LeasesResourceLock,
 			ResourceNamespace: util.NamespaceKarmadaSystem,
+			LeaseDuration:     metav1.Duration{Duration: 15 * time.Second},
+			RenewDeadline:     metav1.Duration{Duration: 10 * time.Second},
+			RetryPeriod:       metav1.Duration{Duration: 2 * time.Second},
 		},
+		BindAddress:                        "0.0.0.0",
+		SecurePort:                         10357,
```
Comment (lonelyCZ) [inline cmd/agent/app/options/options.go:128]: Perhaps, we can rank them by ordered alphabets.

Comment (Zhuzhenghao) [reply cmd/agent/app/options/options.go:128]: ok

**Comment by lonelyCZ** [comment]
Thanks @Zhuzhenghao 

/lgtm

**Comment by carlory** [comment]
cc @RainbowMango

**Comment by RainbowMango** [comment]
> I move all default options to function NewOptions ，so we can get all default values easily

Do we have trouble getting default values?

Sorry, I can't see many benefits from it. In addition,
> fs.StringVar(&o.BindAddress, "bind-address", o.BindAddress,

From this, I feel confused about taking `o.BindAddress` as the default value.

**Comment by carlory** [comment]
@RainbowMango 

In karmada, we can see the setting of the default value in two places, one in the NewOptions function, one in AddFlags, this is very confusing, but also to maintain the consistency of the two settings. 

We should only Set the default value in the NewOptions function, and set the parameter name and parameter description in AddFlags. This will also benefit the reading of the code. The name of the parameter can usually be seen from the field name. Compared with the parameter name, More often than not, we care more about the setting of parameters, and NewOptions is easier to retrieve and maintain than AddFlags.

FYI (kubernetes): 

https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-apiserver/app/options/options.go#L93

FYI (karmada): 

- https://github.com/karmada-io/karmada/blob/HEAD/cmd/scheduler/app/options/options.go#L74
- https://github.com/karmada-io/karmada/blob/HEAD/cmd/scheduler/app/options/options.go#L90

**Comment by lonelyCZ** [comment]
I just found that it seems not convenient for new code readers. If they want to know the meanings of options, they still need to see the AddFlags.

Currently, we don't set more default values in NewOptions() that seems only  for initialization, it seems ok to maintain current uniformal usages that put the defalut in AddFlags().

**Comment by RainbowMango** [comment]
> I just found that it seems not convenient for new code readers. If they want to know the meanings of options, they still need to see the AddFlags.

Yes, that's exactly my concern at https://github.com/karmada-io/karmada/pull/2878#issuecomment-1358742669.

**Comment by Zhuzhenghao** [comment]
Thank you for your feedback on my pull request. I understand the reason for refusal and agree with the decision.Thank you for your help and guidance.

**Pull request closed without merging**